File: programs/protocol/src/data/forwarder.rs
use anchor_lang::prelude::*;

use crate::Name;

#[account(zero_copy)]
#[derive(Debug, Default, PartialEq, Eq)]
pub struct Forwarder {
    pub creator: Pubkey,
    pub authority: Pubkey,
    pub name: Name,
    pub count: u32,
}

File: app/src/utils/parse/forwarder.ts
import { decodeName } from '$sdk/sdk';
import type { ApiForwarder, Forwarder } from '../account/forwarder';

export function parseForwarderToApiForwarder(forwarderAccount: Forwarder): ApiForwarder {
	return {
		...forwarderAccount,
		authority: forwarderAccount.authority.toString(),
		creator: forwarderAccount.creator.toString(),
		name: decodeName(forwarderAccount.name)
	};
}
